internal/serviceapi/orgserviceaccountapi/resource_schema.go

```diff
@@ -0,0 +1,98 @@
+// Code generated by terraform-provider-mongodbatlas using `make generate-resource`. DO NOT EDIT.
+
+package orgserviceaccountapi
+
+import (
+	"context"
+
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+)
+
+func ResourceSchema(ctx context.Context) schema.Schema {
+	return schema.Schema{
+		Attributes: map[string]schema.Attribute{
+			"client_id": schema.StringAttribute{
+				Computed:            true,
+				MarkdownDescription: "The Client ID of the Service Account.",
+			},
+			"created_at": schema.StringAttribute{
+				Computed:            true,
+				MarkdownDescription: "The date that the Service Account was created on. This parameter expresses its value in the ISO 8601 timestamp format in UTC.",
+			},
+			"description": schema.StringAttribute{
+				Required:            true,
+				MarkdownDescription: "Human readable description for the Service Account.",
+			},
+			"name": schema.StringAttribute{
+				Required:            true,
+				MarkdownDescription: "Human-readable name for the Service Account. The name is modifiable and does not have to be unique.",
+			},
+			"org_id": schema.StringAttribute{
+				Required:            true,
+				MarkdownDescription: "Unique 24-hexadecimal digit string that identifies the organization that contains your projects. Use the [/orgs](#tag/Organizations/operation/listOrganizations) endpoint to retrieve all organizations to which the authenticated user has access.",
+			},
+			"roles": schema.ListAttribute{
```

Comment:
currently defined as a List but the underlying implementation of the API stores this property as a set (does not respect order and validates duplicate elements).
CLOUDP-348768 created for defining uniqueElements facet in the API Spec, as per current autogeneration logic this will make it a SetAttribute.